akka-http/src/main/scala/akka/http/scaladsl/server/directives/PathDirectives.scala

```diff
@@ -196,6 +199,40 @@ trait PathDirectives extends PathMatchers with ImplicitPathMatcherConstruction w
       } else pass
     }
 
+  /**
+   * Tries to match the inner route and if it fails with an empty rejection, it tries it again
+   * adding (or removing) the trailing slash on the given path.
+   *
+   * @group path
+   */
+  def ignoreTrailingSlash: Directive0 = {
+    import RouteConcatenation._
+    def flipTrailingSlash(path: Path): Path = {
+      if (path.endsWithSlash && path != Path.SingleSlash) path.reverse.tail.reverse
+      else path ++ Path.SingleSlash
+    }
+
+    // This is not working. Apparently `pass` can't be recovered
+    //    pass.recoverPF {
+    //      case Nil ⇒
+    //        println("this is not called")
+    //        mapRequestContext(req ⇒ req.withUnmatchedPath(flipTrailingSlash(req.unmatchedPath)))
+    //    }
```

Comment:
aha! that was the catch! I wanted to avoid using `handleRejections` because I'm not sure if it might interfere with custom ones users might create along the way inside the `ignoreTrailingSlash`